mathcomp/ssreflect/plugin/trunk/ssreflect.ml4

```diff
@@ -1238,19 +1240,20 @@ let interp_search_notation loc tag okey =
   let err msg = CErrors.user_err ~loc ~hdr:"interp_search_notation" msg in
   let mk_pntn s for_key =
     let n = String.length s in
-    let s' = String.make (n + 2) ' ' in
+    let s' = Bytes.make (n + 2) ' ' in
     let rec loop i i' =
       if i >= n then s', i' - 2 else if s.[i] = ' ' then loop (i + 1) i' else
       let j = try String.index_from s (i + 1) ' ' with _ -> n in
       let m = j - i in
       if s.[i] = '\'' && i < j - 2 && s.[j - 1] = '\'' then
         (String.blit s (i + 1) s' i' (m - 2); loop (j + 1) (i' + m - 1))
       else if for_key && is_ident (String.sub s i m) then
-         (s'.[i'] <- '_'; loop (j + 1) (i' + 2))
+         (Bytes.set s' i' '_'; loop (j + 1) (i' + 2))
       else (String.blit s i s' i' m; loop (j + 1) (i' + m + 1)) in
     loop 0 1 in
-  let trim_ntn (pntn, m) = String.sub pntn 1 (max 0 m) in
+  let trim_ntn (pntn, m) = Bytes.sub pntn 1 (max 0 m) in
```

Comment:
Couldn't you have a simpler patch if you converted back to string before returning in `mk_pntn`? I have the impression that all the other operations below could remain on strings only (do you modify a `ntn` after it has been returned by `mk_ntn`?).